CanaryValidator/Canary.Tests.ps1

```diff
@@ -390,7 +390,85 @@ while ($runCount -le $NumberOfIterations)
                 $asTenantSubscription | Select-AzureRmSubscription -ErrorAction Stop
             }           
         } 
+	
+	Invoke-Usecase -Name 'RoleAssignmentAndCustomRoleDefinition' -Description "Assign a reader role and create a custom role definition" -UsecaseBlock `
+        {
+            $readerRoleName = (Get-AzureRmRoleDefinition -Name Reader).Name
+            $subscriptionID = (Get-AzureRmSubscription -SubscriptionName $tenantSubscriptionName).SubscriptionId
+            $servicePrincipal = (Get-AzureRmADServicePrincipal)[0]
+            $customRoleName = "CustomCanaryRole-" + [Random]::new().Next(1,99)
+
+            Invoke-Usecase -Name 'ListAssignedRoles' -Description "List assigned roles to Service Principle - $($servicePrincipal.DisplayName)" -UsecaseBlock `
+            {
+	            Get-AzureRmRoleAssignment -ObjectId $servicePrincipal.Id -ErrorAction Stop
+            }
+
+            Invoke-Usecase -Name 'AssignReaderRole' -Description "Assign Reader role to Service Principle - $($servicePrincipal.DisplayName)" -UsecaseBlock `
+            {
+	            New-AzureRmRoleAssignment -Scope "/Subscriptions/$subscriptionID" -RoleDefinitionName $readerRoleName -ObjectId $servicePrincipal.Id -ErrorAction Stop
+                $assignedRoles = Get-AzureRmRoleAssignment -RoleDefinitionName $readerRoleName -Scope "/Subscriptions/$subscriptionID" -ErrorAction Stop
+                if (-not $assignedRoles)
+                {
+                    throw [System.Exception] "Unable to assign role ($readerRoleName) to Service Principle ($servicePrincipal.Id) for subscription $tenantSubscriptionName"
+                }                    
+            }
+
+            Invoke-Usecase -Name 'ListRoleDefinitions' -Description "List Role Definitions" -UsecaseBlock `
+            {
+	            Get-AzureRmRoleDefinition -ErrorAction Stop
```

Comment:
For this Get operation and the one below, do a basic check to make sure you receive some expected content back. For example, you can check that Owner and Reader roles are returned.